tarofreighter/chain_porter.go

```diff
@@ -466,9 +467,26 @@ func (p *ChainPorter) transferReceiverProof(pkg *sendPackage) error {
 				ctx, *pkg.ReceiverAddr, receiverProof,
 			)
 			if err != nil {
-				log.Errorf("unable to deliver proof: %v", err)
+				err = fmt.Errorf("unable to deliver proof: %w",
+					err)
+				resultChan <- err
+				return
 			}
+
+			resultChan <- nil
 		}()
+
+		// Wait for the proof delivery to complete, or for the porter
+		// to shut down.
+		select {
+		case <-p.Quit:
```

Comment:
So we're creating a `goroutine` above but then block on the return of it? Why do we need the goroutine in the first place?
Since we're using `p.WithCtxQuitNoTimeout()` the `DeliverProof` should abort on shutdown since the context is canceled. If that doesn't work, then something down the line doesn't listen on the context cancel, which we should fix instead.